akka-remote/src/main/scala/akka/remote/RemoteWatcher.scala

```diff
@@ -202,14 +202,25 @@ private[akka] class RemoteWatcher(
     }
   }
 
+  /** Returns true if either has cluster or `akka.remote.use-unsafe-remote-features-without-cluster`
+   * is enabled. Can be overridden when using RemoteWatcher as a superclass.
+   */
+  @InternalApi protected def shouldWatch(@unused watchee: InternalActorRef): Boolean = {
+    // In this it is unnecessary if only created by RARP, but cluster needs it.
+    // Cleaner than overriding Cluster watcher addWatch/removeWatch just for one boolean test
+    remoteProvider.hasClusterOrUseUnsafe
+  }
+
   def addWatch(watchee: InternalActorRef, watcher: InternalActorRef): Unit = {
     assert(watcher != self)
-    log.debug("Watching: [{} -> {}]", watcher, watchee)
-    watching.addBinding(watchee, watcher)
-    watchNode(watchee)
-
-    // add watch from self, this will actually send a Watch to the target when necessary
-    context.watch(watchee)
+    if (shouldWatch(watchee)) {
+      watching.addBinding(watchee, watcher)
+      watchNode(watchee)
+
+      // add watch from self, this will actually send a Watch to the target when necessary
+      context.watch(watchee)
+      log.debug("Watching: [{} -> {}]", watcher, watchee)
```

Comment:
might be better to move that log up again, since the other statements might generate other logs that are related

Reply:
Sure, but then we log a watch occurring before it is true.

Reply:
that is nitpicking, it's better to see that it is about to watching and then the other messages follow after that.